src/CMakeLists.txt

```diff
@@ -797,13 +806,25 @@ target_compile_options(lvgl PRIVATE
         $<$<COMPILE_LANGUAGE:ASM>: -MP -MD -x assembler-with-cpp>
         )
 
+# LITTLEFS_SRC
+add_library(littlefs STATIC ${LITTLEFS_SRC})
```

Comment:
If possible : find a way to disable some warning generated by littlefs source code. Ex:

```
/home/jf/git/PineTime/src/libs/littlefs/lfs.c:429:13: warning: 'lfs_mlist_isopen' defined but not used [-Wunused-function]
  429 | static bool lfs_mlist_isopen(struct lfs_mlist *head,
```

I'm not sure if cmake provides a way to disable warnings coming from system/external libraries?

Reply:
I tried adding to `-Wno-` + warning name under a specific subcomponents' CXX flags but that didn't have any effect.

Maybe reducing the amount of warnings enabled by the generic variable and then re-enabling in specific sub-blocks?

Reply:
I added `-Wno-unused-function` after `${COMMON_FLAGS}` for the littlefs flags and that disabled the warning.

Beyond that it might be worth splitting out the warning related flags from `COMMON_FLAGS` into a separate list that can be omitted for compiling third party sources where we can't easily fix the warnings.

Reply:
You can set the includes as `SYSTEM` in CMake to silence many of the warnings, too.

Reply:
> I added `-Wno-unused-function` after `${COMMON_FLAGS}` for the littlefs flags and that disabled the warning.
> 
> Beyond that it might be worth splitting out the warning related flags from `COMMON_FLAGS` into a separate list that can be omitted for compiling third party sources where we can't easily fix the warnings.

Thank you, I will review the recommendations.